sklearn/neighbors/_classification.py

```diff
@@ -174,6 +174,8 @@ def fit(self, X, y):
         self : KNeighborsClassifier
             The fitted k-nearest neighbors classifier.
         """
+        _check_weights(self.weights)
```

Comment:
Maybe

```suggestion
        self.weights = _check_weights(self.weights)
```
to be safe. Currently it indeed doesn't do anything, but anyone adds any kind of check there (e.g. a check_array) it's not obvious that the return value is ignored.